From 0ef0b9b2386aeaed53a774ae10df10b8e08478a1 Mon Sep 17 00:00:00 2001
From: Thomas Bernard <miniupnp@free.fr>
Date: Sat, 27 Jul 2024 15:38:47 +0200
Subject: [PATCH] miniupnpd: fix "make install" on BSD

1. default man directory to /usr/share/man
2. miniupnpd.conf and minupnpd.8 are in $(SRCDIR)

fixes #765
---
 miniupnpd/Makefile.bsd | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/miniupnpd/Makefile.bsd b/miniupnpd/Makefile.bsd
index 45be4b45..cde1e964 100644
--- a/Makefile.bsd
+++ b/Makefile.bsd
@@ -1,4 +1,4 @@
-# $Id: Makefile.bsd,v 1.107 2024/06/22 16:48:53 nanard Exp $
+# $Id: Makefile.bsd,v 1.109 2024/07/27 13:36:50 nanard Exp $
 # MiniUPnP project
 # http://miniupnp.free.fr/ or https://miniupnp.tuxfamily.org/
 # Author: Thomas Bernard
@@ -127,6 +127,9 @@ LIBS += -lssl -lcrypto
 INSTALLBINDIR = $(PREFIX)/sbin
 INSTALLETCDIR = $(PREFIX)/etc
 MANPREFIX ?= $(PREFIX)
+.if $(MANPREFIX) == ""
+MANPREFIX = /usr/share
+.endif
 INSTALLMANDIR = $(MANPREFIX)/man
 
 all:	$(EXECUTABLES)
@@ -150,7 +153,7 @@ install:	miniupnpd genuuid
 	$(INSTALL) -d $(DESTDIR)$(INSTALLETCDIR)
 	$(INSTALL) -b miniupnpd.conf $(DESTDIR)$(INSTALLETCDIR)
 	$(INSTALL) -d $(DESTDIR)$(INSTALLMANDIR)
-	$(INSTALL) -m 644 miniupnpd.8 $(DESTDIR)$(INSTALLMANDIR)/man8/miniupnpd.8
+	$(INSTALL) -m 644 $(SRCDIR)/miniupnpd.8 $(DESTDIR)$(INSTALLMANDIR)/man8/miniupnpd.8
 
 # genuuid is using the uuid cli tool available under OpenBSD 4.0 in
 # the uuid-1.5.0 package
@@ -163,7 +166,7 @@ UUID != if which uuidgen 2>&1 > /dev/null; then \
         fi
 
 genuuid:
-	$(MV) miniupnpd.conf miniupnpd.conf.before
+	$(CP) $(SRCDIR)/miniupnpd.conf miniupnpd.conf.before
 	sed -e "s/^uuid=[-0-9a-fA-F]*/uuid=$(UUID)/" miniupnpd.conf.before > miniupnpd.conf
 	$(RM) miniupnpd.conf.before
 
